python/pyarrow/tests/test_serialization.py

```diff
@@ -1093,6 +1093,17 @@ def test_tensor_alignment():
         assert y.ctypes.data % 64 == 0
 
 
+def test_empty_tensor():
+    # ARROW-8122, serialize and deserialize empty tensors
+    x = np.array([], dtype=np.float64)
+    y = pa.deserialize(pa.serialize(x).to_buffer())
+    np.testing.assert_array_equal(x, y)
+
+    x = np.array([[], [], []], dtype=np.float64)
+    y = pa.deserialize(pa.serialize(x).to_buffer())
+    np.testing.assert_array_equal(x, y)
```

Comment:
One more question: can you also test `x.T` ?

Reply:
Done.